src/output.rs

```diff
@@ -440,3 +440,51 @@ impl PartialEq<Output> for WeakOutput {
         self.upgrade().map(|o| &o == other).unwrap_or(false)
     }
 }
+
+/// Source for determining output mode information.
+#[derive(PartialEq, Clone, Debug)]
+pub enum OutputModeSource {
+    /// Automatic mode based on an [`Output`].
+    Auto(Output),
+    /// Static output mode.
+    Static {
+        /// Size of the static output
+        size: Size<i32, Physical>,
+        /// Scale of the static output
+        scale: utils::Scale<f64>,
+        /// Transform of the static output
+        transform: Transform,
+    },
+}
+
+impl TryFrom<&OutputModeSource> for (Size<i32, Physical>, utils::Scale<f64>, Transform) {
```

Comment:
Let me know if my rust-fu betrayed me, but I don't think there's a way to automatically implement this for both `&OutputModeSource` and `OutputModeSource`?

Reply:
`Copy` would probably workaround the ref impl, but OutputModeSource can't because of the `Output`